File: src/ui/main_app/capture-live-monitoring.js
#!/usr/bin/env node
/**
 * Capture screenshots of live agent monitoring in the web UI
 */
const { chromium } = require('playwright');

async function captureLiveMonitoring() {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();

  console.log('📸 Navigating to web UI...');
  await page.goto('http://localhost:3002');
  await page.waitForLoadState('networkidle');

  // Wait a bit for WebSocket connections and initial data
  console.log('⏳ Waiting for live data to load...');
  await page.waitForTimeout(3000);

  // Screenshot 1: Full homepage with SessionMonitor
  console.log('📸 Capturing homepage with SessionMonitor...');
  await page.screenshot({ path: '/tmp/live-monitoring-01-homepage.png', fullPage: true });

  // Try to find activity feed
  console.log('🔍 Looking for activity feed...');
  const feedSelectors = [
    '[data-testid="activity-feed"]',
    '[data-testid="event-list"]',
    'text=Active Sessions',
    'text=Session Monitor'
  ];

  for (const selector of feedSelectors) {
    try {
      const element = await page.locator(selector).first();
      if (await element.isVisible({ timeout: 2000 })) {
        console.log(`✅ Found element with selector: ${selector}`);
        break;
      }
    } catch (e) {
      continue;
    }
  }

  // Screenshot 2: After waiting for more activity
  console.log('⏳ Waiting for more agent activity...');
  await page.waitForTimeout(2000);
  await page.screenshot({ path: '/tmp/live-monitoring-02-activity.png', fullPage: true });

  // Check for agent events
  console.log('🔍 Checking for agent spawn events...');
  try {
    const events = await page.locator('text=/spawn|started|initialized|codex|claude/i').all();
    console.log(`✅ Found ${events.length} agent-related events`);
  } catch (e) {
    console.log('⚠️ No agent events found yet');
  }

  // Screenshot 3: Scroll to see more activity
  console.log('📸 Capturing full page with all activity...');
  await page.evaluate(() => window.scrollTo(0, document.body.scrollHeight));
  await page.waitForTimeout(1000);
  await page.screenshot({ path: '/tmp/live-monitoring-03-scrolled.png', fullPage: true });

  // Get page content to inspect structure
  console.log('📄 Analyzing page structure...');
  const content = await page.content();

  // Look for SessionMonitor or monitoring components
  if (content.includes('SessionMonitor') || content.includes('Active Sessions')) {
    console.log('✅ SessionMonitor component found on page!');
  } else {
    console.log('⚠️ SessionMonitor component not visible');
  }

  if (content.toLowerCase().includes('codex') || content.toLowerCase().includes('claude')) {
    console.log('✅ Agent activity visible in page content!');
  } else {
    console.log('⚠️ No agent activity visible yet');
  }

  // Screenshot 4: Final state
  await page.screenshot({ path: '/tmp/live-monitoring-04-final.png', fullPage: true });

  await browser.close();

  console.log('\n' + '='.repeat(70));
  console.log('✅ SCREENSHOTS CAPTURED');
  console.log('='.repeat(70));
  console.log('📁 /tmp/live-monitoring-01-homepage.png');
  console.log('📁 /tmp/live-monitoring-02-activity.png');
  console.log('📁 /tmp/live-monitoring-03-scrolled.png');
  console.log('📁 /tmp/live-monitoring-04-final.png');
  console.log('='.repeat(70));
}

captureLiveMonitoring().catch(console.error);

File: examples/capture_live_monitoring.py
#!/usr/bin/env python3
"""
Capture screenshots of live agent monitoring in the web UI
"""
from playwright.sync_api import sync_playwright
import time

def capture_live_monitoring():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        print("📸 Navigating to web UI...")
        page.goto('http://localhost:3002')
        page.wait_for_load_state('networkidle')

        # Wait a bit for WebSocket connections and initial data
        print("⏳ Waiting for live data to load...")
        time.sleep(3)

        # Screenshot 1: Full homepage with SessionMonitor
        print("📸 Capturing homepage with SessionMonitor...")
        page.screenshot(path='/tmp/live-monitoring-01-homepage.png', full_page=True)

        # Try to find and screenshot the activity feed
        print("🔍 Looking for activity feed...")
        feed_selectors = [
            '[data-testid="activity-feed"]',
            '[data-testid="event-list"]',
            'text=Active Sessions',
            'text=Session Monitor'
        ]

        for selector in feed_selectors:
            try:
                element = page.locator(selector).first
                if element.is_visible(timeout=2000):
                    print(f"✅ Found element with selector: {selector}")
                    break
            except:
                continue

        # Screenshot 2: After waiting for more activity
        print("⏳ Waiting for more agent activity...")
        time.sleep(2)
        page.screenshot(path='/tmp/live-monitoring-02-activity.png', full_page=True)

        # Check for agent events
        print("🔍 Checking for agent spawn events...")
        try:
            events = page.locator('text=/spawn|started|initialized|codex|claude/i').all()
            print(f"✅ Found {len(events)} agent-related events")
        except:
            print("⚠️ No agent events found yet")

        # Screenshot 3: Scroll to see more activity
        print("📸 Capturing full page with all activity...")
        page.evaluate("window.scrollTo(0, document.body.scrollHeight)")
        time.sleep(1)
        page.screenshot(path='/tmp/live-monitoring-03-scrolled.png', full_page=True)

        # Get page content to inspect structure
        print("📄 Analyzing page structure...")
        content = page.content()

        # Look for SessionMonitor or monitoring components
        if 'SessionMonitor' in content or 'Active Sessions' in content:
            print("✅ SessionMonitor component found on page!")
        else:
            print("⚠️ SessionMonitor component not visible")

        if 'codex' in content.lower() or 'claude' in content.lower():
            print("✅ Agent activity visible in page content!")
        else:
            print("⚠️ No agent activity visible yet")

        # Screenshot 4: Final state
        page.screenshot(path='/tmp/live-monitoring-04-final.png', full_page=True)

        browser.close()

        print("\n" + "="*70)
        print("✅ SCREENSHOTS CAPTURED")
        print("="*70)
        print("📁 /tmp/live-monitoring-01-homepage.png")
        print("📁 /tmp/live-monitoring-02-activity.png")
        print("📁 /tmp/live-monitoring-03-scrolled.png")
        print("📁 /tmp/live-monitoring-04-final.png")
        print("="*70)

if __name__ == "__main__":
    capture_live_monitoring()
